packages/office-ui-fabric-react/src/components/MarqueeSelection/MarqueeSelection.base.tsx

```diff
@@ -216,22 +217,52 @@ export class MarqueeSelectionBase extends BaseComponent<IMarqueeSelectionProps,
 
           this._preservedIndicies = selection && selection.getSelectedIndices && selection.getSelectedIndices();
         }
+
         // We need to constrain the current point to the rootRect boundaries.
-        const constrainedPoint = this.props.isDraggingConstrainedToRoot
-          ? {
-              x: Math.max(0, Math.min(rootRect.width, this._lastMouseEvent!.clientX - rootRect.left)),
-              y: Math.max(0, Math.min(rootRect.height, this._lastMouseEvent!.clientY - rootRect.top))
-            }
-          : {
-              x: this._lastMouseEvent!.clientX - rootRect.left,
-              y: this._lastMouseEvent!.clientY - rootRect.top
-            };
+        // and constrain dragRect within viewport
+        const constrainedPoint = {
+          x: this._lastMouseEvent!.clientX - rootRect.left,
+          y: this._lastMouseEvent!.clientY - rootRect.top
+        };
+
+        let rectLeft = Math.min(this._dragOrigin.x, constrainedPoint.x);
```

Comment:
super ultra minor nit, if you just call this `left`, and `rectTop` just `top`, you can later just use them like:

`const dragRect = { left, top, width, height };`

Reply:
fixed